x-pack/plugins/lens/public/datasources/form_based/form_based.tsx

```diff
@@ -970,18 +970,40 @@ export function getFormBasedDatasource({
       return Object.values(state.layers).map(({ indexPatternId }) => indexPatternId);
     },
 
-    getDatasourceInfo: (state, references, indexPatterns) => {
+    getDatasourceInfo: async (state, references, dataViewsService) => {
       const layers = references ? injectReferences(state, references).layers : state.layers;
+      const indexPatterns: DataView[] = [];
+      for (const { indexPatternId } of Object.values(layers)) {
+        const dataView = await dataViewsService?.get(indexPatternId);
+        if (dataView) {
+          indexPatterns.push(dataView);
+        }
+      }
       return Object.entries(layers).reduce<DataSourceInfo[]>((acc, [key, layer]) => {
         const dataView = indexPatterns?.find(
           (indexPattern) => indexPattern.id === layer.indexPatternId
         );
 
         const columns = Object.entries(layer.columns).map(([colId, col]) => {
+          let fields;
+          if (hasField(col)) {
+            fields = [col.sourceField];
+            if (
+              isColumnOfType<TermsIndexPatternColumn>('terms', col) &&
+              col.params.secondaryFields
```

Comment:
You can use the `operation.getCurrentFields(column)` method here

Reply:
done